src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs

```diff
@@ -1981,22 +1981,27 @@ public static async ValueTask<FormulaValue> Switch(EvalVisitor runner, EvalVisit
 
         // ForAll([1,2,3,4,5], Value * Value)
         public static async ValueTask<FormulaValue> ForAll(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
```

Comment:
This API is missing a CancellationToken
Shouldn't we start from there and ensure all inner tasks use it ?

Reply:
The token is on runner.CheckCancel(); 
But we should be checking that inside the for-loop (see other uses in that file). 
Ideally, we'd have some form of static analysis to verify this,